QUESTIONS.md

```diff
@@ -0,0 +1,297 @@
+# JS Fundamentals Questions
+
+**Do you know how and when o use let and/or const?**
+
+You use *let* for variables that are gonna change/reassign another value and const when you have a constant value in the variable
+
+*Example with 'const'*
+
+```
+const name = "Daniela"
+
+console.log(name)
+```
+
+*Example with 'let'*
+
+```
+for(let i = 0; i < n; i++) {
+    console.log(i);
+}
+```
+
+**How does the variables scope works?**
+
+Depending how you use the variables in your code is how you have accessibility and visibility of these ones. In other words, is when you can or cannot use/access to a variable, depending if this is in a block or not
+
+*Example using 'let':*
+
+```
+{ 
+    let number = 4;
+}
+
+// Number can't be used here
+```
+
+*Example using 'var':*
+
+```
+{
+    var number = 4
+}
+
+// Number can be used here
+
+```
+
+*Example with local scope:*
+
+```
+{
+    for(let i = 0; i < n; i++) {
+        const number = 10;
+    }
+}
+
+// Number cannot be used here
+```
+
+**When should I use *for each* vs *map*?**
+
```

Comment:
Just keep in mind, we use `forEach` to loop in the array and do some logic inside.
`array.map` is used to returns a copy of the array, and inside we can modify the elements as we need.

Note: `array.forEach` is not recommended if you are gonna call async functions inside